content/java/concepts/array-list/terms/addall/addall.md

```diff
@@ -0,0 +1,73 @@
+---
+Title: '.addAll()'
+Description: 'Adds a collection to an ArrayList.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Arrays'
+  - 'Data Types'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The `.addAll()` method is used to add the contents of a collection to an instance of the [`ArrayList`](https://www.codecademy.com/resources/docs/java/array-list) class.
+
+## Syntax
+
+```pseudo
+arrayListInstance.addAll(index, collection);
+```
+
+The contents of `collection` are inserted into the `arrayListInstance` at the index `index`. The `collection` must be of the same [datatype](https://www.codecademy.com/resources/docs/java/data-types) as `arrayListInstance`.
+
+## Example
+
+The following example creates two `ArrayList`s and inserts one into the other.
+
+```java
+import java.util.ArrayList;
+
+public class testAddAll {
+  public static void main(String[] args) {
+
```

Comment:
Since this example has many lines, I would add comments. I've provided comment suggestions that you can change around to how you see fit.